### PR TITLE
add space group grant

### PR DESCRIPTION
**Pull request by fschade**
extend the opaque map to contain an identifier to see if it is a user or group grant.

### CODE REVIEW

**Review by kobergj — APPROVED** [review]

File: pkg/storage/utils/decomposedfs/spaces.go [hunk 1]
```diff
@@ -803,24 +803,32 @@ func (fs *Decomposedfs) storageSpaceFromNode(ctx context.Context, n *node.Node,
 		return nil, err
 	}
 
-	m := make(map[string]*provider.ResourcePermissions, len(grants))
+	pm := make(map[string]*provider.ResourcePermissions, len(grants))
```
Comment (micbar) [inline pkg/storage/utils/decomposedfs/spaces.go:806]: Sorry for the nitpicking, but I am not happy about these var names. Why not permissionsMap?

File: pkg/storage/utils/decomposedfs/spaces.go [hunk 2]
```diff
@@ -803,24 +803,32 @@ func (fs *Decomposedfs) storageSpaceFromNode(ctx context.Context, n *node.Node,
...
 			id = g.Grantee.GetUserId().OpaqueId
 		default:
 			continue
 		}
 
-		m[id] = g.Permissions
+		pm[id] = g.Permissions
 	}
-	marshalled, err := json.Marshal(m)
+	marshalledG, err := json.Marshal(gm)
```
Comment (micbar) [inline pkg/storage/utils/decomposedfs/spaces.go:822]: same

File: pkg/storage/utils/decomposedfs/spaces.go [hunk 3]
```diff
@@ -803,24 +803,32 @@ func (fs *Decomposedfs) storageSpaceFromNode(ctx context.Context, n *node.Node,
...
 	if err != nil {
 		return nil, err
 	}
+
+	marshalledP, err := json.Marshal(pm)
```
Comment (micbar) [inline pkg/storage/utils/decomposedfs/spaces.go:827]: same

File: pkg/storage/utils/decomposedfs/spaces.go [hunk 4]
```diff
@@ -803,24 +803,32 @@ func (fs *Decomposedfs) storageSpaceFromNode(ctx context.Context, n *node.Node,
 		return nil, err
 	}
 
-	m := make(map[string]*provider.ResourcePermissions, len(grants))
+	pm := make(map[string]*provider.ResourcePermissions, len(grants))
+	gm := make(map[string]struct{})
```
Comment (micbar) [inline pkg/storage/utils/decomposedfs/spaces.go:807]: same

**Review by micbar — APPROVED** [review]
LGTM

**Pull request merged**